### PR TITLE
Delete processor intermediate files

### PR DESCRIPTION
**Pull request by kasbah**
Removes all the temp files and those that get uploaded to s3. Just deleting everything at the end of the processing would have been easier but I decided to keep the removal of files as close as possible to the creation of the files. I think it makes more sense this way because it decouples the tasks from the rest of the app.

### CODE REVIEW

File: processor/src/tasks/processGerbers/index.ts [hunk 1]
```diff
@@ -223,6 +223,9 @@ export default async function processGerbers(
       job.updateProgress({ status: 'failed', file, error, outputDir })
     }
   }
+  await Promise.all(
+    filePaths.concat(plottedGerbers.gerbers).map(fp => fs.rm(fp, { force: true })),
+  )
```
Comment (kasbah) [inline processor/src/tasks/processGerbers/index.ts:228]:
```suggestion
```

File: processor/src/tasks/processGerbers/index.ts [hunk 2]
```diff
@@ -1,7 +1,7 @@
 import globule from 'globule'
 import Jszip from 'jszip'
+import fs from 'node:fs/promises'
```
Comment (AbdulrhmnGhanem) [inline processor/src/tasks/processGerbers/index.ts:3]: Shouldn't these get deleted too? 

https://github.com/kitspace/kitspace-v2/blob/89a686c48bb4595f69bfd728a7e1bfd9bd739371/processor/src/tasks/processGerbers/index.ts#LL11C3-L11C3

Comment (AbdulrhmnGhanem) [reply processor/src/tasks/processGerbers/index.ts:3]: I think each job should delete its `outputFiles` when it's done.

File: processor/src/tasks/processKicadPCB/index.ts [hunk 3]
```diff
@@ -1,15 +1,25 @@
...
+
 async function processKicadPCB(
   job: Job,
-  { inputDir, kitspaceYaml = {}, outputDir }: Partial<JobData>,
+  {
+    inputDir,
+    kitspaceYaml = {},
+    outputDir,
+    tmpDir,
+  }: ProcessKicadPcbData & Partial<JobData>,
```
Comment (AbdulrhmnGhanem) [inline processor/src/tasks/processKicadPCB/index.ts:22]: This type name is confusing

```suggestion
export type tmpDirPath = string
async function processKicadPCB(
  job: Job,
  {
    inputDir,
    kitspaceYaml = {},
    outputDir,
    tmpDir,
  }: tmpDirPath & Partial<JobData>,
```

File: processor/src/tasks/processGerbers/index.ts (hunk 2 above)
Comment (kasbah) [reply processor/src/tasks/processGerbers/index.ts:3]: Oops, I accidentally removed the whole deletion. I've amended the last commit.

File: processor/src/tasks/processKicadPCB/index.ts [hunk 4]
```diff
@@ -56,47 +66,47 @@ async function processKicadPCB(
...
 
     const relativeKicadPcbFile = path.relative(inputDir, kicadPcbFile)
     const inputFiles = { [relativeKicadPcbFile]: { type: 'kicad', side: null } }
+    await fs.rm(layoutSvgPath)
     return { inputFiles, gerbers }
   } catch (error) {
     for (const file of filePaths) {
       job.updateProgress({ status: 'failed', file, error, outputDir })
     }
+    await fs.rm(layoutSvgPath, { force: true })
```
Comment (AbdulrhmnGhanem) [inline processor/src/tasks/processKicadPCB/index.ts:82]: This should be in the `finally` block like the `processPCB`.

File: processor/src/tasks/processGerbers/index.ts [hunk 5]
```diff
@@ -223,6 +223,8 @@ export default async function processGerbers(
       job.updateProgress({ status: 'failed', file, error, outputDir })
     }
   }
+
+  await Promise.all(filePaths.map(fp => fs.rm(fp, { force: true })))
```
Comment (AbdulrhmnGhanem) [inline processor/src/tasks/processGerbers/index.ts:227]: `top-meta.png` isn't in `filePaths`
image

File: processor/src/tasks/processKicadPCB/index.ts (hunk 4 above)
Comment (kasbah) [reply processor/src/tasks/processKicadPCB/index.ts]: Changed in 179f9599d9dbbc1

File: processor/src/tasks/processGerbers/index.ts (hunk 5 above)
Comment (kasbah) [reply processor/src/tasks/processGerbers/index.ts]: Fixed with 378da7b7c2b8195

**Review by AbdulrhmnGhanem — APPROVED** [review]

**Pull request merged**